cmake/globals.cmake

```diff
@@ -50,8 +50,27 @@ endif()
 
 set(TEST_CONFIGS_DIR "${CMAKE_BINARY_DIR}/test_configs")
 
-# osquery versions
-set(OSQUERY_VERSION 3.4.0)
+# Detect osquery version through git
+set(OSQUERY_VERSION 0.0.0)
+find_package(Git REQUIRED)
+
+if(GIT_FOUND)
+  execute_process(
+    COMMAND "${GIT_EXECUTABLE}" describe --abbrev=0 --tags
```

Comment:
What is the desired behavior for dirty builds?

I usually use `git describe --tags  --always --dirty` but I'm not sure that's appropriate here.

Reply:
You make a valid point here, it might be better to have a more detailed version when the commit is not at the tag, so we avoid errors when releasing manually and in general it's clearer.
The packages correctly pick up the complete version too in their names.

I wonder if it would make sense to let OSQUERY_VERSION to be set during the configuration phase so that if someone, for their own use, wants to set that specific version it can.
Like in the case of an end user applying a patch/PR on top the "official" release.

It's also true that package names can be renamed after they've been created, to have a cleaner version... hum.

Reply:
Will having extraneous stuff break anything? If not, I'd probably suggest `git describe --tags --dirty`

That will result in the tag, if you're at the tag. And otherwise will append the number of comments, and optionally, `-dirty`. Eg:
* `0.1.0` for a tag
* `0.1.0-1-gb57e` for 1 commit, at sha gb57e, clean working dir
* `0.1.0-1-gb57e-dirty` for 1 commit, at sha gb57e, dirty working dir

Partly I'm thinking about how to detect people coming for help with strange builds. 

Alternately, this is all a bit over-thought, we should take what you wrote, and add a gitsha field later

Reply:
FYI, we use the OSQUERY_VERSION env variable during build.  As a vendor, we have releases independent of the osquery releases.